src/Integration/LocalServices/RuleSetSerializer.cs

```diff
@@ -6,19 +6,67 @@
 //-----------------------------------------------------------------------
 
 using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Xml;
 
 namespace SonarLint.VisualStudio.Integration
 {
     internal sealed class RuleSetSerializer : IRuleSetSerializer
     {
-        public RuleSet LoadRuleSet(string path)
+        private readonly IServiceProvider serviceProvider;
+
+        public RuleSetSerializer(IServiceProvider serviceProvider)
         {
-            return RuleSet.LoadFromFile(path);
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            this.serviceProvider = serviceProvider;
         }
 
-        public void WriteRuleSetFile(RuleSet ruleSet, string path)
+        public RuleSet LoadRuleSet(string ruleSetPath)
         {
-            ruleSet.WriteToFile(path);
+            if (string.IsNullOrWhiteSpace(ruleSetPath))
+            {
+                throw new ArgumentNullException(nameof(ruleSetPath));
+            }
+
+            var fileSystem = this.serviceProvider.GetService<IFileSystem>();
+            fileSystem.AssertLocalServiceIsNotNull();
+
+            if (fileSystem.FileExist(ruleSetPath))
+            {
+                try
+                {
+                    return RuleSet.LoadFromFile(ruleSetPath);
+                }
+                catch (Exception ex) when (ex is InvalidRuleSetException || ex is XmlException || ex is IOException)
+                {
+                    // Log this for testing purposes
+                    Trace.WriteLine(ex.ToString(), nameof(LoadRuleSet));
+                }
+            }
+
+            return null;
+
+        }
+
+        public void WriteRuleSetFile(RuleSet ruleSet, string ruleSetPath)
```

Comment:
Writing is not safe, we're handing and output the exception when it's using (part of binding)